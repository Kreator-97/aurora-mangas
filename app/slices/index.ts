export { default as uiSlice } from './uiSlice'
export { default as authSlice } from './authSlice'
export { default as shoppingCartSlice } from './shoppingCartSlice'
