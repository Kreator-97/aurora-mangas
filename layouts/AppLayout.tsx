import { NextPage } from 'next'
import Head from 'next/head'

import { Navbar, Footer } from '../components'
import { Sidebar } from '../components/sidebar/Sidebar'

interface Props {
  title     : string;
  children  : React.ReactNode;
  maxWidth? : 'sm' | 'md' | 'lg' | 'xl' | 'max'
}

export const AppLayout: NextPage<Props> = ({ title, children, maxWidth = 'xl' }) => {
  return (
    <>
      <Head>
        <title>{title}</title> 
      </Head>
      <Navbar />
      <Sidebar />
      <main
        className='main'
        style={{
          maxWidth: `var(--${maxWidth})`
        }}
      >
        {
          children
        }
      </main>
      <Footer />
    </>
  )
}
