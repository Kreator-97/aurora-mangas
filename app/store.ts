import { configureStore } from '@reduxjs/toolkit'
import { uiSlice, authSlice, shoppingCartSlice } from './slices'

export const store = configureStore({
  reducer: {
    ui: uiSlice,
    auth: authSlice,
    cart: shoppingCartSlice
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
