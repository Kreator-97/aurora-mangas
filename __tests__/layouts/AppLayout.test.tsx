import { render, screen } from '@testing-library/react'
import { AppLayout } from '../../layouts/AppLayout'
import { useSession } from 'next-auth/react'
import { Provider } from 'react-redux'
import { store } from '../../app/store'

jest.mock('next-auth/react')
jest.mock('next/router', () => {
  return {
    useRouter: () => ({query: {q: ''}})
  }
})

describe('tests on AppLayout component', () => {
  (useSession as jest.Mock).mockReturnValue({
    status: 'unauthenticated',
    data: null,
  })

  test('should to match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <AppLayout title='testing' children={<></>} />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should to render children prop', () => {
    render(
      <Provider store={store}>
        <AppLayout title='testing' children={
          <section data-testid="content">
            <h2>Content</h2>
          </section>} />
      </Provider>
    )

    expect(screen.getByTestId('content')).toBeInTheDocument
  })
})
