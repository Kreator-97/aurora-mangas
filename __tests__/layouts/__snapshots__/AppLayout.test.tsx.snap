// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on AppLayout component should to match with snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="content"
    >
      <svg
        color="white"
        cursor="pointer"
        data-testid="menu"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0.5"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 150px; height: 30px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Aurora Mangas Logo"
          class="cursor-pointer"
          data-nimg="fixed"
          data-testid="app-logo"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="relative cursor-pointer"
        data-testid="cart-menu"
      >
        <svg
          color="white"
          cursor="pointer"
          data-testid="shopping-cart-menu"
          fill="currentColor"
          height="32"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="w-6 h-6 text-center font-semibold bg-accentDark bg-opacity-95 rounded-full absolute -top-2 -right-2"
        >
          0
        </span>
      </div>
      <form
        class="flex col-span-2"
        data-testid="search-form"
      >
        <input
          class="w-full rounded border-cyan-400 border p-1 text-sm text-dark"
          data-testid="search-input"
          placeholder="Encuentra tu manga favorito. Ejemplo: one punch man #1"
          type="search"
          value=""
        />
      </form>
      <svg
        color="white"
        cursor="pointer"
        data-testid="search-icon"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16l2.879-2.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242zM21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </nav>
  <div
    class="min-h-screen max-w-[360px] bg-dark fixed top-0 left-0 w-full z-50 p-4 transition-transform -translate-x-full z-[1000]"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <div
        class="w-8"
      />
      <h2
        class="text-2xl"
      >
        Bienvenido
      </h2>
      <svg
        class="cursor-pointer"
        data-testid="close-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="text-center mb-2"
    >
      Ingresa para gestionar tu cuenta
    </p>
    <div
      class="flex gap-2 mb-4"
    >
      <button
        class="btn bg-accent w-full"
      >
        Ingresar
      </button>
      <button
        class="btn ghost w-full"
      >
        Crear cuenta
      </button>
    </div>
  </div>
  <div
    class="fixed top-0 right-0 min-h-screen h-screen min-w-[360px] bg-dark z-50 transition-transform translate-x-full"
  >
    <div
      class="flex justify-between items-center border-b border-strokeLight p-4"
      style="height: 65px;"
    >
      <svg
        cursor="pointer"
        data-testid="close-shopping-cart-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <h2
        class="text-center text-2xl"
      >
        Carrito de compra
      </h2>
      <div
        class="w-7"
      />
    </div>
    <div
      class="flex flex-col gap-4 items-center overflow-y-scroll cartlist-items"
      style="padding-top: 1rem; padding-bottom: 1rem;"
    >
      <h2
        class="text-center"
      >
        No hay elementos en el carrito de compra
      </h2>
    </div>
  </div>
  <main
    class="main"
    style="max-width: var(--xl);"
  />
  <footer
    class="min-h-[150px] border-t border-solid border-stroke"
  >
    <h2
      class="text-xl md:text-3xl text-center p-4 mt-8"
    >
      SÃ­guenos en nuestras redes sociales
    </h2>
    <div
      class="flex justify-center gap-8"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="facebook"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 36px; height: 36px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="twitter"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </footer>
</div>
`;
